src/python/pants/backend/python/goals/tailor.py

```diff
@@ -90,6 +91,30 @@ def is_entry_point(content: bytes) -> bool:
     return _entry_point_re.search(content) is not None
 
 
+@rule_helper
+async def _find_resource_py_typed_targets(
+    py_typed_files_globs: PathGlobs, all_owned_sources: AllOwnedSources
+) -> list[PutativeTarget]:
+    """Find resource targets that may be created after discovering any `py.typed` files."""
+    all_py_typed_files = await Get(Paths, PathGlobs, py_typed_files_globs)
+    unowned_py_typed_files = set(all_py_typed_files.files) - set(all_owned_sources)
+    classified_unowned_py_typed_files = {ResourceTarget: unowned_py_typed_files}
+
+    putative_targets = []
+    for tgt_type, paths in classified_unowned_py_typed_files.items():
+        for dirname, filenames in group_by_dir(paths).items():
+            putative_targets.append(
+                PutativeTarget.for_target_type(
+                    tgt_type,
+                    kwargs={"source": "py.typed"},
+                    path=dirname,
+                    name=None,
```

Comment:
I am not sure if we want it to be `"py_typed"` for all targets, like `resource(name="py_typed", source="py.typed")` we have in https://github.com/pantsbuild/pants/blob/d56d7ef459a99c79acea7fca12cb7f6c786e62a0/src/python/pants/testutil/BUILD#L31

Reply:
We almost certainly want to explicitly name this target. The directory name is expected to be picked up by the default `python_sources()` in the directory, if any.

Reply:
Yes, I suspected this. Done.